### PR TITLE
🐛Bump CAPI to v1.0.4

### LINKED ISSUE #1132: clusterctl in release-0.5 CI fails with cert-manager error
/kind bug

e.g. https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1130/pull-cluster-api-provider-openstack-e2e-test/1490695117076434944/build-log.txt

```
INFO: clusterctl init --core cluster-api --bootstrap kubeadm --control-plane kubeadm --infrastructure openstack
...
action failed after 10 attempts: failed to get cert-manager object /, Kind=, /: Object 'Kind' is missing in 'unstructured object has no kind'
  occurred
```

Issue is [documented upstream](https://cluster-api.sigs.k8s.io/user/troubleshooting.html#failed-clusterctl-init---failed-to-get-cert-manager-object) with fix backported to 1.0.x branch.

### PR DESCRIPTION
**Pull request by mdbooth**
Fixes #1132 

/hold

### CODE REVIEW

File: test/e2e/data/e2e_conf.yaml [hunk 1]
```diff
@@ -12,11 +12,11 @@
 managementClusterName: capo-e2e
 
 images:
-- name: gcr.io/k8s-staging-cluster-api/cluster-api-controller:v1.0.0
+- name: gcr.io/k8s-staging-cluster-api/cluster-api-controller:v1.0.4
```
Comment (sbueringer) [inline test/e2e/data/e2e_conf.yaml:15]: Shouldn't be the cause of the issue but those versions should also be updated in the makefile in the `test-e2e-image-prerequisites` target

**Comment by mdbooth** [comment]
Looks like we may also need https://github.com/kubernetes-sigs/cluster-api/pull/6080.

File: test/e2e/data/e2e_conf.yaml [hunk 2]
```diff
@@ -58,8 +58,8 @@ providers:
 - name: kubeadm
   type: ControlPlaneProvider
   versions:
-  - name: v1.0.0
-    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.0/control-plane-components.yaml"
+  - name: v1.0.4
+    value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.0.4/control-plane-components.yaml"
     type: url
     files:
```
Comment (sbueringer) [inline test/e2e/data/e2e_conf.yaml:64]: You have to change the CAPO version below to v0.5.99. Essentially that is a "dev version" which matches the latest release series of your metadata.yaml

Comment (sbueringer) [reply test/e2e/data/e2e_conf.yaml:64]: You should also drop the 1.0 series in `../data/shared/v1beta1_provider/metadata.yaml` (like in the top-level metadata.yaml)

**Comment by mdbooth** [comment]
We're also waiting on https://github.com/kubernetes-sigs/cluster-api/pull/6080 to merge. When it does we need to bump the `sigs.k8s.io/cluster-api/test` dep to include the fix.

**Review by chrischdi — COMMENTED** [review]
one nit: as @sbueringer mentioned:

We should also update the images here:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/Makefile#L137

**Comment by chrischdi** [comment]
> one nit: as @sbueringer mentioned:
> 
> We should also update the images here: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/Makefile#L137

adding hold for you @mdbooth to decide to either fix the Makefile in this PR or unlock CI and fix the Makefile in a separate PR

/lgtm
/hold

**Pull request merged**